### ISSUE TITLE
Download highest quality songs

### ISSUE THREAD

**Issue by max5354** [z-huang/music#89]
### Checklist

- [X] I've checked that there is no other issue about this feature request.
- [X] This issue contains only one feature request.
- [X] The title of this issue accurately describes the feature request.

### Feature description

add "download highest quality songs" option so we can download 256kbps aac songs.

### Why do you want this feature?

It's necessary

**Comment by z-huang** [comment]
By default, you download the highest quality song that YouTube provides.
I will add a setting option to let users choose what quality(low/medium/high) of audio they would like to listen and download.

**Comment by z-huang** [comment]
I found that 256kbps audio formats are only available for YouTube Premium users. So, #88 is a requirement for this issue.

**Comment by max5354** [comment]
> I found that 256kbps audio formats are only available for YouTube Premium users. So, #88 is a requirement for this issue.

Look some users want to listen 256kbps songs like me. And you're also right that google is data octopus who want our data. So some people want login/signin option but some people don't need this (login/signin) option. So the solution is that you will add warning that "if you turned on signin/Login option than you putting your data in danger" this type of warning. So some people like this option and some people don't like this. It's her/him choice to enable login/signin option. The main benefit of this feature is we will get 256kbps songs. Sorry for my bad English. But above is only solution for all people who wants login/signin option and for those also who don't want that feature.

**Comment by max5354** [comment]
> > I found that 256kbps audio formats are only available for YouTube Premium users. So, #88 is a requirement for this issue.
> 
> Look some users want to listen 256kbps songs like me. And you're also right that google is data octopus who want our data. So some people want login/signin option but some people don't need this (login/signin) option. So the solution is that you will add warning that "if you turned on signin/Login option than you putting your data in danger" this type of warning. So some people like this option and some people don't like this. It's her/him choice to enable login/signin option. The main benefit of this feature is we will get 256kbps songs. Sorry for my bad English. But above is only solution for all people who wants login/signin option and for those also who don't want login/signin feature.

And also if you add that signin/Login option. Than your app will become one of the popular apps. Because no other app is giving 256kbps audio output like 
ymusic (128kbps output)
YouTube vanced (128kbps output)
Libretube (128kbps)
Vuetube (128kbps)
Blackhole(256kbps output but songs link expiring after 6 hour)
YouTube music (256kbps output but we need to pay money)

So this is the benefit of adding login/signin option and also we can play any songs at 256kbps.😊😊😊

**Comment by max5354** [comment]
> I found that 256kbps audio formats are only available for YouTube Premium users. So, #88 is a requirement for this issue.

One last question. Are you planning to add signin/Login option? It will be great if you will add it. 😀😀😀😀

**Comment by z-huang** [comment]
But the fact is that people who have YouTube Premium will use the official YouTube Music app instead.

**Comment by max5354** [comment]
> But the fact is that people who have YouTube Premium will use the official YouTube Music app instead.

You're right. But know one will pay money for youtube premium if you provide premium features for free.

**Comment by z-huang** [comment]
Did you got that 256kbps audio is only for people who log in and have YouTube Premium? If you don't have YT Premium, you won't get the 256kbps audio.

**Comment by max5354** [comment]
> Did you got that 256kbps audio is only for people who log in and have YouTube Premium? If you don't have YT Premium, you won't get the 256kbps audio.

But in YouTube music vanced we can play song at 256kbps without paying single penny

**Comment by z-huang** [comment]
How to know that?

**Comment by max5354** [comment]
> How to know that?

Google said that YouTube music's "high quality" option have 256kbps bitrate....

![20220624_082958](https://user-images.githubusercontent.com/90886938/175453588-3fc8ef7e-0a88-41be-ada9-b53ca6d9ef4a.jpg)
![20220624_082902](https://user-images.githubusercontent.com/90886938/175453603-789f5715-8e7b-41c4-ab59-271d55c3c7ee.jpg)

**Comment by max5354** [comment]
> > How to know that?
> 
> Google said that YouTube music's "high quality" option have 256kbps bitrate....
> 
> ![20220624_082958](https://user-images.githubusercontent.com/90886938/175453588-3fc8ef7e-0a88-41be-ada9-b53ca6d9ef4a.jpg) ![20220624_082902](https://user-images.githubusercontent.com/90886938/175453603-789f5715-8e7b-41c4-ab59-271d55c3c7ee.jpg)

So now you believe? That youtube music's audio quality is 256kbps for premium users. And i am using youtube music vanced (currently dead). So the option is you can take code of 256kbps audio quality from youtube music vanced. Just like YouTube music vanced app you can also add 256kbps audio quality in your music app. Vanced said that youtube vanced and youtube music vanced's "project code" available at github. So you can take a youtube music vanced app's code and add that "code" in your music app.

**Comment by max5354** [comment]
> I'm still a little bit in doubt about this. I think this should be controlled by the server side.
> 
> Vanced doesn't provide source code of the app, since YT is closed-source. If you know where is the code, please put a link. If there's any app that can detect the bitrate of the playing song, we will be more clear on this question.

I don't have knowledge about coding but i think this will helpful for you 
https://github.com/cuynu/ytvanced

**Comment by z-huang** [comment]
> I don't have knowledge about coding but i think this will helpful for you
https://github.com/cuynu/ytvanced

This is just a repo storing APKs.🥲
